### ISSUE TITLE
Depixelated resized images

### ISSUE THREAD

**Issue by seularts** [niklasvh/html2canvas#1304]
I have tested with the latest non-minified version of html2canvas (V1 Alpha 3): http://tzr.ro/x/p/index2.php

I don't see anything wrong with the code in the Console. This code is as simple as it can get with a PNG file:

```
<body>
<img src="logo.png" height="100">
<script src="https://ajax.googleapis.com/ajax/libs/jquery/3.2.1/jquery.min.js"></script>
<script type="text/javascript" src="html2canvas.js"></script>
<script>
html2canvas(document.body).then(function(canvas) {
    document.body.appendChild(canvas);
});
</script>
</body>
```

**Comment by seularts** [comment]
I have done a series of tests with various resizing scripts and as far as I can tell, all render the image depixelted due to generating the code of the image outside the browser engine. Therefore the browser renders an image smoother when it is resized and the pixels are compacted. So far, I haven't seen any script that can do the same! Maybe there is a way for your script to use the browser engine to create smooth images as well!? This is just a deduction, so I can't be 100% sure this is the cause!

**Comment by seularts** [comment]
I might have found a solution! I used a SVG file through css to make it as a font that way it acts as a webfont at this point (I used this site to generate it: https://icomoon.io ). This seems to have fixed the scaling issue. Also, another fix I found for the underlines is to use div:before with content as dots.

`.line:before {
    content: "..................................................................................................";
    display: block;
    font-size: 60px;
    font-family: Georgia;
    color: #fff;
    max-width: 100%;
    overflow: hidden;
	margin-top: -40px;
}`

It's a bit messy but works!

**Comment by seularts** [comment]
Seems that the only viable solution for colored big-sized images is to manually shrink them to the required height and width in order to render well in the final snapshot. Sadly there is no capable script out there to do this automatically without loss of resolution in the process!
